sycl/source/detail/program_impl.cpp

```diff
@@ -416,6 +416,11 @@ RT::PiKernel program_impl::get_pi_kernel(const string_class &KernelName) const {
           Err);
     }
     Plugin.checkPiResult(Err);
+
+    // Some PI Plugins (like OpenCL) require this call to enable USM
```

Comment:
NIT:  if a kernel uses only accessors, will this call lead to unnecessary overhead during its invocation? If not known, maybe a TODO to check this could be added.